### PR TITLE
Adopt latest roslyn analyzers

### PR DESCRIPTION
**Pull request by jcouv**
I'm holding off using `#nullable enable` in any project because there is a bug in that analyzer. After adding nullable annotations to the public API files, the analyzer complains with RS0016 and RS0017...

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -27,7 +27,7 @@
   </PropertyGroup>
   <PropertyGroup>
     <!-- Versions used by several individual references below -->
-    <RoslynDiagnosticsNugetPackageVersion>2.9.6</RoslynDiagnosticsNugetPackageVersion>
+    <RoslynDiagnosticsNugetPackageVersion>3.0.0-beta2.20059.3+77df2220</RoslynDiagnosticsNugetPackageVersion>
```
Comment (jcouv) [inline eng/Versions.props]: @dotnet/roslyn-infrastructure @jaredpar  I've update the version number for roslyn analyzer package. I'm seeing the new analyzers kick in within VS, but it looks like `build.cmd` is still using some older version (2.9.6 as far as I can tell). 
Is that expected? What's the process from here?

Comment (mavasani) [reply eng/Versions.props]: @jcouv - is command line build using older version of the analyzer package OR compiler toolset?

File: .editorconfig [hunk 2]
```diff
@@ -205,3 +205,7 @@ csharp_space_between_square_brackets = false
 csharp_prefer_braces = true:silent
 csharp_preserve_single_line_blocks = true
 csharp_preserve_single_line_statements = true
+
+# error RS0037: PublicAPI.txt is missing '#nullable enable'
```
Comment (mavasani) [inline .editorconfig]: Will this be done as a separate PR or with further commits in this PR?

File: src/Test/Utilities/Portable/Diagnostics/CommonDiagnosticAnalyzers.cs [hunk 3]
```diff
@@ -572,7 +572,9 @@ public override void Initialize(AnalysisContext context)
         public sealed class AnalyzerWithCSharpCompilerDiagnosticId : DiagnosticAnalyzer
         {
             public static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
+#pragma warning disable RS1029 // Do not use reserved diagnostic IDs.
```
Comment (mavasani) [inline src/Test/Utilities/Portable/Diagnostics/CommonDiagnosticAnalyzers.cs]: Might want to turn off RS1029 for all non-shipping projects: https://github.com/dotnet/roslyn/blob/master/eng/config/rulesets/NonShipping.ruleset

File: .editorconfig (hunk 2 above)
Comment (sharwell) [inline .editorconfig]:
```suggestion
# warning RS0037: PublicAPI.txt is missing '#nullable enable'
```

File: src/Test/Utilities/Portable/Diagnostics/CommonDiagnosticAnalyzers.cs (hunk 3 above)
Comment (sharwell) [reply src/Test/Utilities/Portable/Diagnostics/CommonDiagnosticAnalyzers.cs]: The targeted suppression seems fine for now, considering it was scope-limited

File: eng/Versions.props (hunk 1 above)
Comment (jcouv) [reply eng/Versions.props]: @mavasani Not sure how to tell the difference. Probably both. 
Assuming that analyzer package is referenced by compiler toolset (correct me if that's wrong), there is no compiler toolset which references the new analyzer package that I need (since it's brand new).

File: .editorconfig (hunk 2 above)
Comment (jcouv) [reply .editorconfig]: Each project needs to decide whether they want to enable this. But they cannot enable this right now, because of the issue above (two versions of analyzers involved).
